apps/frontend/src/hooks/aave/useAaveRates.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { formatUnits } from 'ethers/lib/utils';
+
+import { Decimal } from '@sovryn/utils';
+
+import { RAY_DECIMALS } from '../../utils/math';
+import { useAaveReservesData } from './useAaveReservesData';
+import { calculateUtilizationRate } from './utils';
+
+export interface RatesDataResult {
+  currentUsageRatio: Decimal;
+  optimalUsageRatio: Decimal;
+  baseVariableBorrowRate: Decimal;
+  variableRateSlope1: Decimal;
+  variableRateSlope2: Decimal;
+  underlyingAsset: string;
+  name: string;
+  symbol: string;
+  decimals: number;
+}
+
+export const useAaveInterestRatesData = (
+  symbol: string,
+): {
+  data: RatesDataResult | null;
+  error: string | null;
+} => {
+  const [data, setData] = useState<RatesDataResult | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const { reserves } = useAaveReservesData();
+  const reserve = useMemo(
+    () =>
+      reserves.find(
+        r => r.symbol.toLocaleLowerCase() === symbol.toLocaleLowerCase(),
+      ),
+    [reserves, symbol],
+  );
+
+  useEffect(() => {
```

Comment:
you can use the same memo for finding and formatting the the reserve. Using an useEffect will cause an extra render